### PR TITLE
hide rsv on manage devices

### LINKED ISSUE #6495: Hide RSV from Manage devices dropdown on Facility edit page
### Background
We want to have RSV exist as a supported disease in our database, but not allow users to see it as a supported disease for single-entry. Any devices they add to a facility should not appear to support RSV until then.

### Action requested
Hide RSV as a test type on the Device dropdown on the Facility edit page. If a device only supports RSV, do not show it in the dropdown. Gate this with a new feature flag for _single-entry_ RSV.

Image

### Acceptance Criteria
Even if a device supports RSV in our backend, RSV should not appear in the test type column for that device in the dropdown. If a device only supports RSV, it should not appear in the dropdown at all.

### Additional context
Potentially a bug or some inconsistent behavior across browsers?: #6493
Kinda blocked by #6445 - will be easier to test and know what to filter out exactly once that ticket is done

### PR DESCRIPTION
**Pull request by fzhao99**
# FRONTEND PULL REQUEST

## Related Issue

- Fixes #6495 

## Changes Proposed

- Adds new single entry RSV feature flag
- Filters out RSV-only devices completely and removes RSV from supported diseases on manage facility device page
- I left out UI tests and only tested the underlying filtering logic so we don't have tests randomly breaking once we turn the flag off but if folks want happy to go back and add

## Testing

- Turn singleEntryRsv feature flag on / off and make sure display updates correctly

#### RSV Included (rsv filtered out, other diseases are included)
https://github.com/CDCgov/prime-simplereport/assets/29645040/bb2577a9-5444-43f6-96eb-3b58b8e51284

#### RSV Only (complete filter)
https://github.com/CDCgov/prime-simplereport/assets/29645040/e286aa07-7308-4047-9dbf-04aea8d530d0

## On
https://github.com/CDCgov/prime-simplereport/assets/29645040/f16a1859-3ab4-47e0-8e9d-4be50cb3dc6b

### CODE REVIEW

File: frontend/src/app/Settings/Facility/Components/ManageDevices.tsx [hunk 1]
```diff
@@ -16,6 +18,41 @@ interface Props {
   onChange: (selectedItems: string[]) => void;
 }
 
+type SupportedDisease = {
```
Comment (fzhao99) [inline frontend/src/app/Settings/Facility/Components/ManageDevices.tsx:21]: the prop type was typed as any, so adding a utility type here so TS doesn't yell at me. If anyone knows of an existing type that I can steal instead, lmk.

Happy also to add a tech debt ticket to fix this particular typing inconsistency frontend-wide if folks think it's worth it.

File: frontend/src/app/Settings/Facility/Components/ManageDevices.tsx [hunk 2]
```diff
@@ -16,6 +17,45 @@ interface Props {
...
+    (sd) => sd.supportedDisease.name
+  );
+  if (supportedDiseases.length === 1 && supportedDiseases[0] === "RSV") {
+    return [];
+  }
+  if (supportedDiseases.includes("RSV")) {
+    return supportedDiseaseArray.filter(
+      (d) => d.supportedDisease.name !== "RSV"
+    );
+  }
```
Comment (BobanL) [inline frontend/src/app/Settings/Facility/Components/ManageDevices.tsx:44]: Wouldn't lines 40-44 also take care of the case of a single item in the array?

File: frontend/src/app/Settings/Facility/Components/ManageDevices.tsx [hunk 3]
```diff
@@ -16,6 +17,42 @@ interface Props {
...
+  const supportedDiseases = supportedDiseaseArray.map(
+    (sd) => sd.supportedDisease.name
+  );
+  if (supportedDiseases.includes("RSV")) {
+    return supportedDiseaseArray.filter(
+      (d) => d.supportedDisease.name !== "RSV"
+    );
+  }
+  return supportedDiseaseArray;
+}
```
Comment (BobanL) [inline frontend/src/app/Settings/Facility/Components/ManageDevices.tsx:43]: Wouldn't the fileter also take care of the case of a single item in the array?

**Review by emyl3 — COMMENTED** [review]
There is an edge case if we...
- enable feature flag and add RSV only device to a facility
- disable feature flag
- remove all the other devices from that facility so you are left with no visible devices for that facility
- can still save/edit the form (even though it shows an error on first save) because technically the facility has a device even though visually it looks like the facility has no devices

Not sure how likely this edge case is but wanted to flag this.

**Comment by fzhao99** [comment]
> There is an edge case if we...
> 
> * enable feature flag and add RSV only device to a facility
> * disable feature flag
> * remove all the other devices from that facility so you are left with no visible devices for that facility
> * can still save/edit the form (even though it shows an error on first save) because technically the facility has a device even though visually it looks like the facility has no devices
> 
> Not sure how likely this edge case is but wanted to flag this.

Thanks for the callout! I think since this would require both a flipping on / off of the flag and a facility to delete all but an RSV only device, chances of hitting it would pretty low and I think we can just be aware of it as a theoretical possibility. If other folks disagree and want it handled explicitly though we can figure something out. 

cc: @mehansen

**Review by BobanL — APPROVED** [review]
Code looks good. I'm personally ok with the risk of that edge case, but Elisa with the hawk eyes to figure it out.

**Review by emyl3 — APPROVED** [review]
LGTM! 😸

**Pull request merged**